aten/src/ATen/core/ivalue.cpp

```diff
@@ -370,6 +370,8 @@ size_t IValue::hash(const IValue& v) {
       return c10::get_hash(*v.toTuple());
     case Tag::Device:
       return c10::get_hash(v.toDevice());
+    case Tag::ComplexDouble:
+      return c10::hash<c10::complex<double>>()(v.toComplexDouble());
```

Comment:
Hmm, why are you adding ComplexDouble, but not `ComplexFloat` nor `ComplexHalf`)? And how does it align with the goal of PR to make Dict keys to be all ivalues?

Reply:
@malfet  I think such modification can make the Dict class more consistent and coherent in hash, and easy to use the Dict class in C++ language. I really hope that it can be merged. Thanks!